[14.0][FIX] account_financial_report: General ledger changes

victoralmau opened the pull request:
General ledger changes:
- Fix total_bal_curr value from General ledger to prevent wrong values
- Prevent wrong values

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT41041

chienandalu commented on account_financial_report/report/trial_balance.py:
```diff
@@ -226,21 +226,37 @@ def _compute_account_amount(
-        total_amount[acc_id][prt_id]["initial_balance"] = tb["balance"]
-        total_amount[acc_id][prt_id]["ending_balance"] = tb["balance"]
+        # Add keys to dict if not exists
+        if acc_id not in total_amount.keys():
```
The `in` expression over a `dict` already check if the object is a key.
```suggestion
        if acc_id not in total_amount:
```

chienandalu commented on account_financial_report/report/trial_balance.py:
```diff
@@ -226,21 +226,37 @@ def _compute_account_amount(
+        # Add keys to dict if not exists
+        if acc_id not in total_amount.keys():
+            total_amount[acc_id] = self._prepare_total_amount(foreign_currency)
+        if prt_id not in total_amount[acc_id].keys():
```
Same here
```suggestion
        if prt_id not in total_amount[acc_id]:
```

chienandalu reviewed (approved):
Just two little comments. Aside from that LGTM :+1:

pedrobaeza commented on account_financial_report/report/trial_balance.py:
```diff
@@ -226,21 +226,37 @@ def _compute_account_amount(
+            "initial_balance": 0.0,
+            "ending_balance": 0.0,
+        }
+        if foreign_currency:
```
Shouldn't this be initialized always? On v14, `amount_currency` and `currency_id` is always filled in the `account.move.line`.

victoralmau replied:
Ok, changes done.

pedrobaeza replied:
I was just asking. What I don't want also is to have filled such columns when no different currency is set. You tell me...

victoralmau replied:
Before this PR and now, the value of the corresponding columns are filled in (I didn't want to make any "extra" changes to avoid side effects).

Taking into account `amount_currency` and `currency_id` fields is probably interesting, but I prefer not to change the current behaviour.

pedrobaeza replied:
But I'm not saying that. I'm asking if current behavior is correct after the previous changes you did are correct. Is foreign currency being shown in the PDF/XLSX only when it's different from the main currency?

victoralmau replied:
Ok sorry, now I understand what you are saying.

Before if the initial and final balance was shown (in the corresponding currency) but NOT the currency column, I have added a change to show this data.

pedrobaeza commented:
Testing on runbot, I'm afraid this is not working properly.

On Excel export:

![Selección_016](https://user-images.githubusercontent.com/7165771/210821595-42e03a00-52c3-4ae7-8d8c-17c3317e910f.png)

On direct view:

![Selección_017](https://user-images.githubusercontent.com/7165771/210821658-90c8c2fb-3f93-49d7-9e18-f52bab92afcc.png)

victoralmau commented:
> Testing on runbot, I'm afraid this is not working properly.
> 
> On Excel export:
> 
> ![Selección_016](https://user-images.githubusercontent.com/7165771/210821595-42e03a00-52c3-4ae7-8d8c-17c3317e910f.png)
> 
> On direct view:
> 
> ![Selección_017](https://user-images.githubusercontent.com/7165771/210821658-90c8c2fb-3f93-49d7-9e18-f52bab92afcc.png)

This is General Ledger isn't it (I was checking trial balance)?
I'll check it anyway now.

victoralmau committed:
[FIX] account_financial_report: General ledger changes:
- Fix total_bal_curr value from General ledger to prevent wrong values
- Prevent wrong values

TT41041

pedrobaeza commented:
On general ledger, I still find an error on accumulated amounts:

![Selección_020](https://user-images.githubusercontent.com/7165771/211006275-d57c1ef5-3eff-4c7d-b184-280483aabfd9.png)

and on trial balance, no foreign currency is shown at all:

![Selección_021](https://user-images.githubusercontent.com/7165771/211006325-b9c8c7ea-7295-47ff-89e5-7eebb04b0f1f.png)

victoralmau commented:
Problems related to General ledger is because `bal_curr` = `amount_currency` is set (`account.move.line `field) but in v14 that value is always set even if currency is not set, and that causes the "inconsistency" you are seeing.

pedrobaeza commented:
Yes, but you need to fix them. It's not correct as it is.

victoralmau commented:
> On direct view:
> 
> ![Selección_017](https://user-images.githubusercontent.com/7165771/210821658-90c8c2fb-3f93-49d7-9e18-f52bab92afcc.png)

Previously, the value was:
ending balance = initial balance - balance
175.74 = 541.10 - 378.89

I don't understand why that was the case, but I BELIEVE it is incorrect and should be similar to General ledger.

Changes have been applied to show the correct values (with currency) and to remove the currency column.

victoralmau commented:
@pedrobaeza can you review it again?

Now, only minimal changes related to general ledger are added. 
Trial balance changes are at https://github.com/OCA/account-financial-reporting/pull/992 (it is a FWP of v13).

pedrobaeza reviewed (approved):
/ocabot merge patch
